pkg/scheduler/core/generic_scheduler.go

```diff
@@ -146,6 +152,24 @@ func (g *genericScheduler) snapshot() error {
 // If it succeeds, it will return the name of the node.
 // If it fails, it will return a FitError error with reasons.
 func (g *genericScheduler) Schedule(ctx context.Context, prof *profile.Profile, state *framework.CycleState, pod *v1.Pod) (result ScheduleResult, err error) {
+	annotation, ok := pod.Annotations["trace.kubernetes.io/context"]
+	if ok {
+		propagator := trace.DefaultHTTPPropagator()
+		reqHeader := &http.Header{}
+		decodedContextBytes, err := base64.StdEncoding.DecodeString(annotation)
+		if err != nil {
+			return result, err
```

Comment:
do you intent to interrupt the business logic based on parsing of trace context results?

Reply:
True, this probably does not need to result in cancelled scheduling.